ui/src/groups/Join/JoinGroupModal.tsx

```diff
@@ -31,21 +32,25 @@ export default function JoinGroupModal() {
             >
               Back
             </button>
-            {gang.invite ? (
+            {gang.invite && status !== 'loading' ? (
               <button
                 className="button bg-red text-white dark:text-black"
                 onClick={reject}
               >
                 Reject Invite
               </button>
             ) : null}
-            <button
-              className="button ml-2 bg-blue text-white dark:text-black"
-              onClick={button.action}
-              disabled={button.disabled}
-            >
-              {button.text}
-            </button>
+            {status === 'loading' ? (
+              <LoadingSpinner />
```

Comment:
I would add some text here and also add sizing to the spinner so that it doesn't accidentally get sized wrong from other changes. Also error state